src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -86,18 +87,22 @@ public InnReachResponseDTO trackShippedItem(String trackingId, String centralCod
     var innReachTransaction = getTransaction(trackingId, centralCode);
 
     var itemBarcode = itemShipped.getItemBarcode();
+    var callNumber = itemShipped.getCallNumber();
 
     var transactionPatronHold = (TransactionPatronHold) innReachTransaction.getHold();
-    transactionPatronHold.setShippedItemBarcode(itemBarcode);
 
-    if (Objects.nonNull(itemBarcode)) {
-      var itemByBarcode = inventoryService.getItemByBarcode(itemBarcode);
-      if (Objects.nonNull(itemByBarcode)) {
-        transactionPatronHold.setShippedItemBarcode(itemBarcode + transactionPatronHold.getItemAgencyCode());
+    if (nonNull(itemBarcode)) {
+      var itemByBarcode = inventoryService.findItemByBarcode(itemBarcode);
+      if (itemByBarcode.isPresent()) {
+        transactionPatronHold.setFolioItemBarcode(itemBarcode + transactionPatronHold.getItemAgencyCode());
+        updateFolioAssociatedItem(transactionPatronHold.getFolioItemId(), itemBarcode + transactionPatronHold.getItemAgencyCode());
       }
+      transactionPatronHold.setShippedItemBarcode(itemBarcode);
     }
```

Comment:
`folioItemBarcode` should be set too when `itemByBarcode` is not present, it will just have the same value as `shippedItemBarcode` in this case

Reply:
The `updateFolioAssociatedItem` method should be called in any case too. Because when we create inventory virtual items for patron hold, they come without barcodes. This story is about setting it.

Reply:
@oliinyko , yes) you're right, thanks!